internal/db/schema/migrations/oss/postgres/101/01_app_token_base.up.sql

```diff
@@ -0,0 +1,246 @@
+
+  -- Create trigger for app_token deletion tracking
+  create trigger insert_deleted_id after delete on app_token
+    for each row execute function insert_deleted_id('app_token_deleted');
+
+  -- Create index on delete_time for app_token_deleted
+  create index app_token_deleted_delete_time_idx on app_token_deleted (delete_time);
+
+  create table app_token_permission (
+    private_id wt_private_id primary key,
+    app_token_id wt_public_id not null,
+      constraint app_token_permission_app_token_id_idx
+        foreign key (app_token_id) references app_token(public_id)
+        on delete cascade
+        on update cascade
+  );
+  comment on table app_token_permission is
+    'app_token_permission is the base table for application token permissions.';
+
+  -- Create index on app_token_id for efficient token-based lookups
+  create index app_token_permission_app_token_id_idx on app_token_permission (app_token_id);
+
+  create function insert_app_token_permission_subtype() returns trigger
+  as $$
+  begin
+    insert into app_token_permission
+      (private_id, app_token_id)
+    values
+      (new.private_id, new.app_token_id);
+    return new;
+  end;
+  $$ language plpgsql;
+  comment on function insert_app_token_permission_subtype() is
+    'insert_app_token_permission_subtype is used to automatically insert a row into the app_token_permission table '
+    'whenever a row is inserted into the subtype table';
+
+  create table app_token_permission_grant (
+    permission_id wt_private_id
+      constraint app_token_permission_grant_fkey
+      references app_token_permission(private_id)
+        on delete cascade
+        on update cascade,
+    canonical_grant wt_canonical_grant not null
+      constraint app_token_permission_iam_grant_fkey
+        references iam_grant(canonical_grant)
+        on delete cascade
+        on update cascade,
+    raw_grant text not null
+      constraint raw_grant_must_not_be_empty
+      check(
+        length(trim(raw_grant)) > 0
+      ),
+    primary key(permission_id, canonical_grant)
+  );
+  comment on table app_token_permission_grant is
+    'app_token_permission_grant contains grants assigned to app tokens in project scope';
+
+  create trigger upsert_canonical_grant_trigger before insert on app_token_permission_grant
+    for each row execute procedure upsert_canonical_grant();
+
+  create table app_token_cipher (
+    app_token_id wt_public_id primary key
+      constraint app_token_cipher_app_token_fkey
+        references app_token(public_id)
+          on delete cascade
+          on update cascade,
+    key_id text not null
+      constraint kms_data_key_version_fkey
+        references kms_data_key_version (private_id)
+        on delete restrict
+        on update cascade,
+    token bytea not null unique
+  );
+  comment on table app_token_cipher is
+    'app_token_cipher is the table for application token encryption keys. '
```

Comment:
Assuming the intent here was to say "app token encryption keys" instead of "application token encryption keys":

```suggestion
    'app_token_cipher is the table for app token encryption keys. '
```

Reply:
Maybe not -- feel free to ignore this if "application tokens" was intended